jws/jws-mediator/src/main/java/org/wso2/financial/services/apim/mediation/policies/jws/JwsRequestHeaderValidationMediator.java

```diff
@@ -0,0 +1,596 @@
+ * KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.wso2.financial.services.apim.mediation.policies.jws;
+
+import com.nimbusds.jose.JOSEException;
+import com.nimbusds.jose.JWSHeader;
+import com.nimbusds.jose.JWSObject;
+import com.nimbusds.jose.JWSVerifier;
+import com.nimbusds.jose.jwk.ECKey;
+import com.nimbusds.jose.jwk.JWK;
+import com.nimbusds.jose.jwk.JWKSet;
+import com.nimbusds.jose.jwk.RSAKey;
+import com.nimbusds.jose.util.Base64URL;
+import org.apache.commons.lang3.StringUtils;
+import org.apache.commons.logging.Log;
+import org.apache.commons.logging.LogFactory;
+import org.apache.synapse.MessageContext;
+import org.apache.synapse.SynapseException;
+import org.apache.synapse.core.axis2.Axis2MessageContext;
+import org.apache.synapse.mediators.AbstractMediator;
+import org.json.JSONArray;
+import org.json.JSONObject;
+import org.wso2.financial.services.apim.mediation.policies.jws.constants.JwsConstants;
+import org.wso2.financial.services.apim.mediation.policies.jws.utils.JwsUtils;
+
+import java.io.IOException;
+import java.net.URL;
+import java.nio.charset.StandardCharsets;
+import java.security.PublicKey;
+import java.security.cert.X509Certificate;
+import java.text.ParseException;
+import java.time.Instant;
+import java.time.LocalDateTime;
+import java.time.ZoneId;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+import java.util.stream.Collectors;
+
+import javax.naming.InvalidNameException;
+import javax.naming.ldap.LdapName;
+import javax.naming.ldap.Rdn;
+import javax.ws.rs.HttpMethod;
+
+/**
+ * Mediator for validating JWS request headers.
+ */
+public class JwsRequestHeaderValidationMediator extends AbstractMediator {
+
+    private static final Log log = LogFactory.getLog(JwsRequestHeaderValidationMediator.class);
+
+    private String jwSignatureHeaderName;
+    private String validTrustAnchor = null;
+    private String jwsSupportedAlgorithms = null;
+
+    /**
+     * JWK retrieved from JwksURI matching to the KeyID.
+     */
+    private JWK matchingJWK = null;
+
+    @Override
+    public boolean mediate(MessageContext messageContext) {
+
+        org.apache.axis2.context.MessageContext axis2MessageContext = ((Axis2MessageContext) messageContext)
+                .getAxis2MessageContext();
+        Map<String, String> headers = (Map<String, String>)
+                axis2MessageContext.getProperty(org.apache.axis2.context.MessageContext.TRANSPORT_HEADERS);
+
+        String payload = JwsUtils.extractPayload(axis2MessageContext, headers);
+        String jwSignature = headers.get(jwSignatureHeaderName);
+
+        if (StringUtils.isEmpty(jwSignature)) {
+            String errorDescription = "Empty JWS Signature";
```

Comment:
shall we include the request  url also for log message

Reply:
Fixed with 55d7e83f22cd8b61c36a2ec94553099c99069c10